src/Microsoft.SqlTools.ServiceLayer/DacFx/GenerateTSqlModelOperation.cs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Diagnostics;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.Utility;
+using Microsoft.SqlServer.Dac.Model;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent creating a dacfx model
+    /// </summary>
+    class GenerateTSqlModelOperation
+    {
+        public GenerateTSqlModelParams Parameters { get; }
+
+        public GenerateTSqlModelOperation(GenerateTSqlModelParams parameters)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+        }
+
+        /// <summary>
+        /// Generate model from sql files, if no sql files are passed in then it generates an empty model.
+        /// </summary>
+        public TSqlModel GenerateTSqlModel()
+        {
+            try
+            {
+                TSqlModelOptions options = new TSqlModelOptions();
+                SqlServerVersion version = (SqlServerVersion)Enum.Parse(typeof(SqlServerVersion), Parameters.ModelTargetVersion);
+
+                var model = new TSqlModel(version, options);
+                // read all sql files
+                foreach (string filePath in Parameters.FilePaths)
+                {
+                    string fileContent = System.IO.File.ReadAllText(filePath);
```

Comment:
maybe check if file exists before loading?

Reply:
Generally that's considered an anti-pattern since the file can be deleted between when you check and when you try to read regardless.

And I think it's reasonable to have this entire thing fail if one of the files doesn't exist. We don't want to just skip those because a file not existing implies there's some other issue and could result in an incorrect model.

Reply:
I agree to fail all if one does not exist, I think we should check and add the log for debugging

Reply:
it's up  to you if you want to add extra logging really. it's useful for debugging later. I think this PR is just missing tests. I approve if adding tests later or in separate PR

Reply:
I don't think we should get in the habit of adding extra overhead (checking for a file) just for logging. And the error from trying to read a non-existent file is clear enough as to what the issue is so it's not going to get us anything useful. 

If we're concerned about logging errors then we might as well just add logging to RequestContext.SendError so that we get it for all messages automatically.

Reply:
I'll follow up with adding logging to SendError, that's a good idea and will help out everything in STS.